src/v/features/feature_table.cc

```diff
@@ -232,9 +284,15 @@ feature_table::feature_table() {
             }
         }
     }
+
```

Comment:
declaring the destructor made this true, but I wonder if it's worth being explicit:
```c++
    static_assert(
      !std::is_move_constructible_v<feature_table>
        && !std::is_move_assignable_v<feature_table>
        && !std::is_copy_constructible_v<feature_table>
        && !std::is_copy_assignable_v<feature_table>,
      "The probe captures a reference to this");
```

Reply:
Yeah I like it, I've adopted this now